community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/planner/CostBasedExecutablePlanBuilder.scala

```diff
@@ -69,35 +72,36 @@ case class CostBasedExecutablePlanBuilder(monitors: Monitors,
         val (logicalPlan, pipeBuildContext) = produceLogicalPlan(ast, rewrittenSemanticTable)(planContext)
         monitor.foundPlan(inputQuery.queryText, logicalPlan)
         try {
-          val res = produceCompiledPlan(logicalPlan, inputQuery, rewrittenSemanticTable,
-            planContext, pipeBuildContext, planBuilderMonitor)
+          val res = runtimeName match {
+            case CompiledRuntimeName => produceCompiledPlan(logicalPlan, rewrittenSemanticTable, planContext, planBuilderMonitor)
+            case _ => produceInterpretedPlan(logicalPlan, pipeBuildContext, planContext)
```

Comment:
We should fallback if `useErrorsOverWarnings == false` right? Not sure I agree that it is not part of the PR, the code has been changed from always falling back to sometimes falling back and in the process it looks like we changed the default to being interpreted again?